src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/Extensions/VsLspExtensions_Range.cs

```diff
@@ -0,0 +1,149 @@
+        => (startLine, startCharacter, endLine, endCharacter) = (range.Start.Line, range.Start.Character, range.End.Line, range.End.Character);
+
+    public static LinePositionSpan ToLinePositionSpan(this Range range)
+        => new(range.Start.ToLinePosition(), range.End.ToLinePosition());
+
+    public static bool IntersectsOrTouches(this Range range, Range other)
+    {
+        if (range.IsBefore(other))
+        {
+            return false;
+        }
+
+        if (range.IsAfter(other))
+        {
+            return false;
+        }
+
+        return true;
+    }
+
+    private static bool IsBefore(this Range range, Range other) =>
+        range.End.Line < other.Start.Line || (range.End.Line == other.Start.Line && range.End.Character < other.Start.Character);
+
+    private static bool IsAfter(this Range range, Range other) =>
+        other.End.Line < range.Start.Line || (other.End.Line == range.Start.Line && other.End.Character < range.Start.Character);
+
+    public static bool OverlapsWith(this Range range, Range other)
+    {
+        ArgHelper.ThrowIfNull(range);
+        ArgHelper.ThrowIfNull(other);
+
+        return range.ToLinePositionSpan().OverlapsWith(other.ToLinePositionSpan());
+    }
+
+    public static bool LineOverlapsWith(this Range range, Range other)
+    {
+        ArgHelper.ThrowIfNull(range);
+        ArgHelper.ThrowIfNull(other);
+
+        var overlapStart = range.Start.Line;
+        if (range.Start.Line.CompareTo(other.Start.Line) < 0)
+        {
+            overlapStart = other.Start.Line;
+        }
+
+        var overlapEnd = range.End.Line;
+        if (range.End.Line.CompareTo(other.End.Line) > 0)
+        {
+            overlapEnd = other.End.Line;
+        }
+
+        return overlapStart.CompareTo(overlapEnd) <= 0;
+    }
+
+    public static bool Contains(this Range range, Range other)
+    {
+        ArgHelper.ThrowIfNull(range);
+        ArgHelper.ThrowIfNull(other);
+
+        return range.Start.CompareTo(other.Start) <= 0 && range.End.CompareTo(other.End) >= 0;
+    }
+
+    public static bool SpansMultipleLines(this Range range)
+    {
+        ArgHelper.ThrowIfNull(range);
+
+        return range.Start.Line != range.End.Line;
+    }
+
+    public static bool IsSingleLine(this Range range)
+    {
+        ArgHelper.ThrowIfNull(range);
+
+        return range.Start.Line == range.End.Line;
+    }
+
+    public static bool IsUndefined(this Range range)
+    {
+        ArgHelper.ThrowIfNull(range);
+
+        return range == VsLspFactory.UndefinedRange;
+    }
+
+    public static int CompareTo(this Range range1, Range range2)
+    {
+        var result = range1.Start.CompareTo(range2.Start);
+
+        if (result == 0)
+        {
+            result = range1.End.CompareTo(range2.End);
+        }
+
+        return result;
+    }
+
+    public static TextSpan AsTextSpan(this Range range, SourceText text)
```

Comment:
Should this be `To` ?

Reply:
I wasn't certain, since that's the name [you originally gave this method](https://github.com/dotnet/razor/commit/72e190ce845f0534b6a81630745c2bfe0200f7c8). 😝 

There's already a `ToTextSpan` with the same arguments, which is probably why `AsTextSpan` was picked in the first place. I'll take a look at how these might be better unified.

Reply:
Or maybe it wasn't your fault. It looks like [you only moved that one](https://github.com/dotnet/razor/pull/10042/commits/72e190ce845f0534b6a81630745c2bfe0200f7c8).

Reply:
Either way, it was before I was educated to the To/As divide 😁

Reply:
Awww... now I feel a little bad: https://github.com/dotnet/razor/pull/4203/commits/c14531780c36b853155d8d039de6afd9206f8c44